drivers/counter/counter_ll_stm32_rtc.c

```diff
@@ -64,7 +65,15 @@ static int rtc_stm32_start(struct device *dev)
 {
 	ARG_UNUSED(dev);
 
+#if defined(CONFIG_SOC_SERIES_STM32WBX) || defined(CONFIG_STM32H7_DUAL_CORE) || defined(CONFIG_SOC_SERIES_STM32MP1X)
```

Comment:
I think it would be better in general to have a define on top like

```c
#define HAS_HSEM (defined(CONFIG_SOC_SERIES_STM32WBX) || defined(CONFIG_STM32H7_DUAL_CORE) || defined(CONFIG_SOC_SERIES_STM32MP1X))
```
 and use:

```c
#if HAS_HSEM
// code for SoC that have HSEM
#endif
```

It makes code more readable and code requires less changes when adding support for new SoC.

Reply:
Fully reworked. Done